### PR TITLE
feat(python of provider): Stop using provider.provider

### PR DESCRIPTION
**Pull request by thomaspoignant**
# Description
Stop using provider.provider as mentioned in https://github.com/open-feature/python-sdk/issues/319

## Next steps:
- When the python SDK is available, bump the SDK version in poetry
- Release a new version.

# Checklist
- [x] I have tested this code
- [x] I have followed the [contributing guide](CONTRIBUTING.md)

### CODE REVIEW

**Comment by thomaspoignant** [comment]
Waiting for the SDK to be released and we will release this PR when available.

**Comment by federicobond** [comment]
FYI provider state should be handled by the Python SDK according to the latest OF spec. There is no longer a need to expose a `get_status` method yourself.

File: openfeature/providers/python-provider/gofeatureflag_python_provider/metadata.py [hunk 1]
```diff
@@ -5,4 +5,7 @@
 
 @dataclass
 class GoFeatureFlagMetadata(Metadata):
+    def __init__(self):
+        pass
+
```
[the remark this answers is not in the document]
Comment (thomaspoignant) [reply openfeature/providers/python-provider/gofeatureflag_python_provider/metadata.py:10]: I've added it because pycharm was complaining 😮‍💨

**Comment by thomaspoignant** [comment]
> FYI provider state should be handled by the Python SDK according to the latest OF spec. There is no longer a need to expose a `get_status` method yourself.

Thanks for the heads up @federicobond, I've removed the usage of get_status from the code.

**Pull request merged**